compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -590,35 +596,36 @@ class MapAllocatorCache {
     }
   }
 
-  void releaseIfOlderThan(CachedBlock &Entry, u64 Time) REQUIRES(Mutex) {
-    if (!Entry.isValid() || !Entry.Time)
-      return;
-    if (Entry.Time > Time) {
-      if (OldestTime == 0 || Entry.Time < OldestTime)
-        OldestTime = Entry.Time;
-      return;
-    }
+  inline void release(CachedBlock &Entry) {
+    DCHECK(Entry.Time != 0);
     Entry.MemMap.releaseAndZeroPagesToOS(Entry.CommitBase, Entry.CommitSize);
     Entry.Time = 0;
   }
 
   void releaseOlderThan(u64 Time) EXCLUDES(Mutex) {
     ScopedLock L(Mutex);
-    if (!EntriesCount || OldestTime == 0 || OldestTime > Time)
+    if (!EntriesCount)
       return;
-    OldestTime = 0;
-    for (uptr I = 0; I < Config::getQuarantineSize(); I++)
-      releaseIfOlderThan(Quarantine[I], Time);
-    for (uptr I = 0; I < Config::getEntriesArraySize(); I++)
-      releaseIfOlderThan(Entries[I], Time);
-  }
 
+    for (uptr I = 0; I < Config::getQuarantineSize(); I++) {
```

Comment:
I'm still considering what we should do with the quarantine here. For now I've left the quarantine release logic untouched, but it doesn't sit right with me that we scan the quarantine even if the quarantine has been disabled.

Reply:
hmmm, I think we should avoid scan the quarantine entries here. I think we can have a function like

```
bool IsQuarantineEnabled() {
  return useMemoryTagging<Config>(Options) && QuarantinePos != -1U;
}
```
to bypass scanning the quarantine entries. It doesn't need to be done in this CL by the way.

Reply:
Ok, I'll make this change in a separate CL. Should I restore the `OldestTime` variable for now. I doubt that removing it would have any noticeable performance difference for scanning the quarantine, but let me know what you think.

Reply:
Yep, I think it's fine to remove it